docs/tables/aws_backup_recovery_point.md

```diff
@@ -63,4 +68,57 @@ from
   aws_backup_recovery_point
 where
   is_encrypted = 1;
+```
+
+### Get associated tags for the targeted Recovery Points EC2, EBS and S3 resource types
+Retrieving metadata, in the form of tags, for recovery points associated with three resource types - EC2 instances, EBS volumes, and S3 buckets. Tags are key-value pairs that provide valuable information about AWS resources.
+
+```sql+postgres
+select
+  r.backup_vault_name as backup_vault_name,
+  r.recovery_point_arn as recovery_point_arn,
+  r.resource_type as resource_type,
+case
+    when r.resource_type = 'EBS' then (
+      select tags from aws_ebs_snapshot where arn = concat(
+        (string_to_array(r.recovery_point_arn, '::'))[1],
+        ':',
+        r.account_id,
+        ':',
+        (string_to_array(r.recovery_point_arn, '::'))[2]
+      )
+    )
+    when r.resource_type = 'EC2' then (
+      select tags from aws_ec2_ami where image_id = (string_to_array(r.recovery_point_arn, '::image/'))[2]
+    )
+    when r.resource_type = 'S3' then (
+      select tags from aws_s3_bucket where arn = r.recovery_point_arn
+    )
+end as target_resource_tags,
```

Comment:
I think there was some confusion. We are interested only in the tags on the RP itself (not on the original backed up resource), so the switch-case for S3 should instead return the tags of the record itself.

To be clear there is a distinction between a "resource" and the "rp (snapshot)" in AWS Backup. The "resource" is what was backed up like an EC2 instance, EBS volume or an S3 bucket, and the RP is its snapshot, which is either an EC2 AMI, or an EBS snapshot, or an S3 RP (fully managed by AWS Backup).

What "Full AWS Backup management" means is that AWS Backup service itself maintains the snapshot and its tags. Non-full management means AWS Backup just orchestrates the snapshots using other service-specific implementations of snapshots and therefore tags on those snapshots need to be fetched using those service-specific APIs.

```suggestion
    when r.resource_type = 'S3' then r.tags
end as tags,
```
You could also add `r.resource_type = 'S3' or r.resource_type = 'EFS'` to cover EFS here as well.


Same thing applies to SQL example below

Reply:
I have updated the query to get the tags for the EFS resource type as well.